pkg/domain/domain.go

```diff
@@ -442,16 +443,23 @@ func (do *Domain) fetchAllSchemasWithTables(m *meta.Meta) ([]*model.DBInfo, erro
 	if err != nil {
 		return nil, err
 	}
+	if len(allSchemas) == 0 {
+		return nil, nil
+	}
+
 	splittedSchemas := do.splitForConcurrentFetch(allSchemas)
-	doneCh := make(chan error, len(splittedSchemas))
+
+	eg, ectx := errgroup.WithContext(context.Background())
```

Comment:
use NewErrorGroupWithRecoverWithCtx?

Reply:
Maybe it isn't need. 
The `WorkerPool` can not run the `ErrorGroupWithRecover`, unless 'WorkerPool` is deleted and use ErrorGroupWithRecover directly.

Reply:
Why do you want to use `NewWorkerPool`? We can use `SetLimit` on errgroup